### DISCUSSION TITLE
Enable Device Login for Unrecognized Device

### DISCUSSION THREAD

**Discussion by Scribing** [dani-garcia/vaultwarden#4104] [Ideas]
While the official device login is only suitable for known devices ([Bitwarden's official blog](https://bitwarden.com/de-DE/blog/access-your-bitwarden-vault-without-a-password/)) ,
it is possible to configure the server to make this feature available to all devices. 
During login, the browser sends a request to `/api/devices/knowndevice` to check if the current device is recognized. 
The button is only displayed if the response is true. 
However, by modifying the response from false to true, you can also log in normally by clicking the "Log in with device" button.

This feature makes it very convenient and secure to log in with a browser in incognito mode, and it does not directly expose the master password through keyboard input.

**Comment by BlackDex** [comment]
It actually makes it less secure. It would allow anyone to login from any device not first logged in by using a master password and MFA.

If you left a device open which is able to confirm the login you just exposed a quick way for someone else to login.

From my perspective we will not add something like this.
If you wish to still do this by your self i suggest to catch the call via your reverse proxy. But i would not recommend this for live usage.

**Reply by Scribing** [reply]
In private deployments, I believe that allowing any user (or some specified users) to see the "Log in with device" option during login will not pose a significant security risk. This is because an attacker would need to:

1. Access the address of the private deployment (within the local area network),
2. Obtain the target user's email address (even incorrect email addresses will be shown as sent),
3. Send a login request (which only has a validity period of 15 minutes),
4. Ensure that the user has enabled the device login feature,
5. Have the user view and confirm the login (which is almost impossible with fingerprint phrase and IP confirmation).

Even now, it is possible to directly send device login requests (though this requires a simple manual modification of the returned content). Therefore, this feature merely controls the visibility of certain UI elements and does not differ from the current level of security. 
It also requires the administrator to modify the config.json to enable it.
